### PR TITLE
Bump Jackson from 2.12.3 to 2.12.4

### PR DESCRIPTION
**Pull request by basil**
Testing Done: `mvn verify` passes with both the default Jenkins version and 2.300

CC @timja @jglick

### CODE REVIEW

**Review by timja — APPROVED** [review]
@dwnusbaum @jtnord

File: src/test/java/com/fasterxml/jackson/dataformat/xml/XmlMapperTest.java [hunk 1]
```diff
@@ -22,11 +20,7 @@ public void smokes() throws Throwable {
     }
 
     private static void _smokes(JenkinsRule r) throws Throwable {
-        XMLInputFactory inputFactory =
-                XMLInputFactory.newFactory(
-                        XMLInputFactory.class.getName(), XmlFactory.class.getClassLoader());
-        XmlFactory factory = new XmlFactory(inputFactory);
-        XmlMapper mapper = new XmlMapper(factory);
```
Comment (bitwiseman) [inline src/test/java/com/fasterxml/jackson/dataformat/xml/XmlMapperTest.java:29]: Why? Was this change required or optional?

Comment (timja) [reply src/test/java/com/fasterxml/jackson/dataformat/xml/XmlMapperTest.java:29]: it was a workaround for a classloading issue in jackson < 2.12.4 (classloading issue still present till jackson 2.13.0 but jackson has included a workaround now)

see https://github.com/jenkinsci/jackson2-api-plugin/pull/82 https://github.com/FasterXML/jackson-dataformat-xml/issues/483 https://github.com/FasterXML/jackson-dataformat-xml/pull/480

**Review by jtnord — APPROVED** [review]

**Review by jglick — APPROVED** [review]

**Comment by basil** [comment]
Happy Friday! Could this please be merged and released?

**Comment by elkh510** [comment]
hi 
Could please some one say when this PR will be merged ?
@jtnord @jglick

**Comment by elops-od** [comment]
Hi,
Could some one tell when this pr is planned to be merged?
@jtnord @jglick pls
Many thanks

**Comment by timja** [comment]
ping
@dwnusbaum @alecharp @jvz @jtnord @rsandell @car-roll @olamy @MRamonLeon @bitwiseman

**Comment by wwuck** [comment]
@olamy The latest github release at https://github.com/jenkinsci/jackson2-api-plugin/releases/tag/jackson2-api-2.12.4 has the incorrect version name `2.2.14`.  I'm guessing this doesn't affect the actual plugin version?

**Comment by olamy** [comment]
@wwuck typo fixed :) thanks
